### PR TITLE
Add sequence id to acknowledgement results key

### PR DESCRIPTION
**Pull request by NeverHappened**
Created with this https://p2pvalidator.atlassian.net/browse/NTRN-221 in mind, although it's more an improvement than fix.
Because audit issue was really non existent - relayer cannot submit same message with same sequence id to the chain.

What's there:
- added sequence id to ack storage - so we can query acks for exact sequence id we issued
- add deposit for .sh scripts (fix)
- sudo call returns error if we try to write ack into same (port_id, seq_id) pair - (NOT SURE ABOUT THAT)

Also: See https://github.com/neutron-org/neutron-integration-tests/pull/21 for related integration tests suite updates

### CODE REVIEW

File: contracts/neutron_interchain_txs/src/msg.rs [hunk 1]
```diff
@@ -16,6 +16,7 @@ pub enum QueryMsg {
     // this query returns acknowledgement result after interchain transaction
     AcknowledgementResult {
         interchain_account_id: String,
+        sequence_id: u64,
```
Comment (foxpy) [inline contracts/neutron_interchain_txs/src/msg.rs:19]: This change will break our integration tests suite here: https://github.com/neutron-org/neutron-integration-tests/blob/main/src/testcases/interchaintx.test.ts#L394

Please open updating PR in integration tests as well.

**Review by foxpy — CHANGES_REQUESTED** [review]

**Review by oopcode — APPROVED** [review]
tested with https://github.com/neutron-org/neutron-integration-tests/pull/21/files, everything's fine; please update the docs and tutorials

**Comment by quasisamurai** [comment]
test w `feat/introduce-sequence-id` works flawless

**Review by quasisamurai — APPROVED** [review]

**Review by swelf19 — APPROVED** [review]

**Review by pr0n00gler — APPROVED** [review]

File: contracts/neutron_interchain_txs/src/contract.rs [hunk 2]
```diff
@@ -295,7 +297,7 @@ fn execute_undelegate(
 }
 
 fn execute_clean_ack_results(deps: DepsMut) -> StdResult<Response<NeutronMsg>> {
-    let keys: Vec<StdResult<String>> = ACKNOWLEDGEMENT_RESULTS
+    let keys: Vec<StdResult<(String, u64)>> = ACKNOWLEDGEMENT_RESULTS
```
Comment (albertandrejev) [inline contracts/neutron_interchain_txs/src/contract.rs:300]: You can use `Vec<_>` instead
```suggestion
    let keys: Vec<_> = ACKNOWLEDGEMENT_RESULTS
```

**Pull request merged**